traitlets/traitlets.py

```diff
@@ -2477,8 +2477,9 @@ def instance_init(self, obj):
 class Dict(Instance):
     """An instance of a Python dict."""
     _trait = None
+    _key_trait = None
 
-    def __init__(self, trait=None, traits=None, default_value=Undefined,
+    def __init__(self, trait=None, key_trait=None, traits=None, default_value=Undefined,
```

Comment:
My preference would be for all of these after the first to be keyword-*only* arguments, but Python 2 doesn't make that very easy other than `**kwargs`. I would add it to the end and treat it as if it were keyword-only.